libs/algorithms/include/hpx/parallel/algorithms/for_each.hpp

```diff
@@ -224,10 +224,10 @@ namespace hpx { namespace parallel { inline namespace v1 {
                 std::forward<Proj>(proj));
         }
         // forward declare the segmented version of for_each_ algorithm
-        template <typename ExPolicy, typename SegIter, typename F,
-            typename Proj>
-        inline typename util::detail::algorithm_result<ExPolicy, SegIter>::type
-        for_each_(ExPolicy&& policy, SegIter first, SegIter last, F&& f,
+        template <typename ExPolicy, typename SegIterB, typename SegIterE,
+            typename F, typename Proj>
+        inline typename util::detail::algorithm_result<ExPolicy, SegIterB>::type
+        for_each_(ExPolicy&& policy, SegIterB first, SegIterE last, F&& f,
```

Comment:
You might want to change the implementation of this function as well (just the prototype) ...

Reply:
@hkaiser Isn't that the prototype? What other part should I change?

Reply:
See here: https://github.com/STEllAR-GROUP/hpx/blob/master/libs/segmented_algorithms/include/hpx/parallel/segmented_algorithms/for_each.hpp#L183-L187